src/libraries/System.Text.Json/Common/JsonSimpleNamingPolicy.cs

```diff
@@ -0,0 +1,147 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Buffers;
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal sealed class JsonSimpleNamingPolicy : JsonNamingPolicy
+    {
+        private readonly bool _lowercase;
+        private readonly char _boundary;
+
+        internal JsonSimpleNamingPolicy(bool lowercase, char boundary) =>
+            (_lowercase, _boundary) = (lowercase, boundary);
+
+        public override string ConvertName(string name)
+        {
+            int bufferLength = name.Length * 2;
+            char[]? buffer = bufferLength > 512
+                ? ArrayPool<char>.Shared.Rent(bufferLength)
+                : null;
+
+            int resultLength = 0;
+            Span<char> result = buffer is null
+                ? stackalloc char[512]
+                : buffer;
+
+            void WriteWord(ref Span<char> result, ReadOnlySpan<char> word)
```

Comment:
Nit: this helper method sits in the middle of the `ConvertName` body and makes it more difficult to review. Please move to the bottom of the outer method or make it a private method of the class itself.

Additionally, this method captures environment other than the parameters. This results in new closure allocations per invocation, so please refactor to a static local and ensure that all dependencies are passed explicitly as parameters.

Reply:
>This results in new closure allocations per invocation

@eiriktsarpalis, afaik that's not true given you couldn't even capture a span in a closure. It's better than F# ;)
As far as I known C# passes captures through additional ref parameters in the final codegen.

Reply:
The closure is a struct, so no allocation, but sure, I will address all changes to improve perf more.